Modules/_uuidmodule.c

```diff
@@ -91,45 +121,1730 @@ py_windows_has_stable_node(void)
+// Bytes 10-15: node                     (48 bits)
+//
+// Note that the time attributes are only relevant to versions 1, 6 and 7.
+//
+// Version field is located in byte 6; most significant 4 bits:
+//
+// Variant field is located in byte 8; most significant variable bits:
+//   0xxx: Reserved for NCS compatibility
+//   10xx: RFC 4122/9562 (standard)
+//   110x: Reserved for Microsoft compatibility
+//   111x: Reserved for future definition
+
+#define RANDOM_BUF_SIZE 256
+#define MAX_FREE_LIST_SIZE 32
+
+/* State of the _uuid module */
+typedef struct {
+    PyTypeObject *UuidType;
+
+    PyObject *safe_uuid;
+
+    PyObject *uint128_max;
+
+    PyObject *random_func;
+    PyObject *random_size_int;
+    PyObject *time_func;
+
+    // A freelist for uuid objects -- 15-20% performance boost.
+    uuidobject *freelist;
+    uint64_t freelist_size;
+
+    // We overfetch entropy to speed up successive uuid generations;
+    // this enables 10x peformance boost.
+    uint8_t random_buf[RANDOM_BUF_SIZE];
+    uint64_t random_idx;
+    uint64_t random_last_pid;
+
+    // UUID v7 state
+    uint64_t last_timestamp_v7;
+    uint64_t last_counter_v7;
+    uint8_t last_timestamp_v7_init;
+} uuid_state;
+
+#include "clinic/_uuidmodule.c.h"
+
+/*[clinic input]
+class _uuid.UUID "uuidobject *" "&UuidType"
+[clinic start generated code]*/
+/*[clinic end generated code: output=da39a3ee5e6b4b0d input=84ae6e2089cffd3f]*/
+
+// Forward declarations
+static int from_hex(uuidobject *self, PyObject *hex);
+static int from_bytes_le(uuidobject *self, Py_buffer *bytes_le);
+static int from_int(uuidobject *self, PyObject *int_value, int validate);
+static int from_fields(uuidobject *self, PyObject *fields);
+static PyObject * get_SafeUUID(uuid_state *state);
+
+static uint64_t
+uuid_getpid(void) {
+    #if !defined(MS_WINDOWS) || defined(MS_WINDOWS_DESKTOP) || defined(MS_WINDOWS_SYSTEM)
+    return (uint64_t)getpid();
 #else
-    ADD_INT("has_stable_extractable_node", 0);
+    return (uint64_t)GetCurrentProcessId();
 #endif
+}
+
+static struct PyModuleDef uuidmodule;
+
+static inline uuid_state *
+get_uuid_state(PyObject *mod)
+{
+    uuid_state *state = PyModule_GetState(mod);
+    assert(state != NULL);
+    return state;
+}
+
+static inline uuid_state *
+get_uuid_state_by_cls(PyTypeObject *cls)
+{
+    PyObject *module = PyType_GetModuleByDef(cls, &uuidmodule);
+    assert(module != NULL);
+    uuid_state *state = get_uuid_state(module);
+    assert(state != NULL);
+    return state;
+}
+
+// Forward declaration
+static PyObject *uuid_from_bytes_array(PyTypeObject *type, uint8_t bytes[16]);
+
+static int
+gen_time(uuid_state *state, PyTime_t* time)
+{
+    if (state->time_func == NULL) {
+        return PyTime_Time(time);
+    }
+
+    PyObject *ret = PyObject_CallNoArgs(state->time_func);
+    if (ret == NULL) {
+        return -1;
+    }
+
+    if (!PyLong_CheckExact(ret)) {
+        PyErr_SetString(PyExc_ValueError, "random_time must return int");
+    }
```

Comment:
```suggestion
    if (!PyLong_CheckExact(ret)) {
        PyErr_SetString(PyExc_ValueError, "random_time must return int");
        Py_DECREF(ret);
        return NULL;
    }
```

You cannot call more from the C API if there is an exception set.